### PR TITLE
AM: fix GetDisplayVersion

### PR DESCRIPTION
**Pull request by CrazyMax**
master:
pre

PR:
post

master:
m1_m1

PR:
m1_p2

### CODE REVIEW

File: src/core/hle/service/am/am.cpp [hunk 1]
```diff
@@ -1353,14 +1353,28 @@ void IApplicationFunctions::GetDisplayVersion(Kernel::HLERequestContext& ctx) {
...
-    FileSys::PatchManager pm{system.CurrentProcess()->GetTitleID()};
-    const auto res = pm.GetControlMetadata();
+    const auto res = [this]() {
+        const auto title_id = system.CurrentProcess()->GetTitleID();
+
+        FileSys::PatchManager pm{title_id};
+        auto res = pm.GetControlMetadata();
+
+        if (res.first != nullptr) {
+            return std::move(res);
```
Comment (lioncash) [inline src/core/hle/service/am/am.cpp:1363]:
```suggestion
            return res;
```

File: src/core/hle/service/am/am.cpp [hunk 2]
```diff
@@ -1353,14 +1353,28 @@ void IApplicationFunctions::GetDisplayVersion(Kernel::HLERequestContext& ctx) {
...
+        }
+
+        // Try with title id when set update bit
+        FileSys::PatchManager pm_update{title_id | 0x800};
+        auto res_update = pm_update.GetControlMetadata();
+        return std::move(res_update);
```
Comment (lioncash) [inline src/core/hle/service/am/am.cpp:1369]:
```suggestion
        return pm_update.GetControlMetadata();
```

File: src/core/hle/service/am/am.cpp [hunk 3]
```diff
@@ -1353,14 +1353,28 @@ void IApplicationFunctions::GetDisplayVersion(Kernel::HLERequestContext& ctx) {
 
     std::array<u8, 0x10> version_string{};
 
-    FileSys::PatchManager pm{system.CurrentProcess()->GetTitleID()};
-    const auto res = pm.GetControlMetadata();
+    const auto res = [this]() {
```
Comment (lioncash) [inline src/core/hle/service/am/am.cpp:1356]:
```suggestion
    const auto res = [this] {
```

**Comment by FernandoS27** [comment]
Thank you very much, could you do the fixes above and add a description comment to your main PR it's for logging reasons.

File: src/core/hle/service/am/am.cpp [hunk 4]
```diff
@@ -1353,14 +1353,26 @@ void IApplicationFunctions::GetDisplayVersion(Kernel::HLERequestContext& ctx) {
...
+        const auto title_id = system.CurrentProcess()->GetTitleID();
+
+        FileSys::PatchManager pm{title_id};
+        auto res = pm.GetControlMetadata();
+        if (res.first != nullptr) {
+            return res;
+        }
+
+        // Try with title id when set update bit
+        FileSys::PatchManager pm_update{title_id | 0x800};
```
Comment (asLody) [inline src/core/hle/service/am/am.cpp:1366]: FileSys::PatchManager pm_update{FileSys::GetUpdateTitleID(title_id)};

**Comment by bunnei** [comment]
LGTM, thank you!

**Pull request merged**